controllers/authorino_controller.go

```diff
@@ -443,73 +478,96 @@ func (r *AuthorinoReconciler) createAuthorinoPermission(authorino *api.Authorino
 	}
 
 	// get ServiceAccount from authorino instance namespace
-	sa := &k8score.ServiceAccount{}
-	saName := resourcePrefixName + "-authorino"
-	saNsdName := namespacedName(authorino.Namespace, saName)
-	if err := r.Get(context.TODO(), saNsdName, sa); err != nil {
-		if errors.IsNotFound(err) {
-			// ServiceAccount doesn't exit - create one
-			ctrl.SetControllerReference(authorino, sa, r.Scheme)
-			sa.Name = saName
-			sa.Namespace = authorino.Namespace
-			if err := r.Client.Create(context.TODO(), sa); err != nil {
-				return r.wrapErrorWithStatusUpdate(
-					logger, authorino, r.setStatusFailed(api.AuthorinoUnableToCreateServiceAccount),
-					fmt.Errorf("failed to create %s ServiceAccount, err: %v", saName, err),
-				)
-			}
+	if sa, err := r.createAuthorinoServiceAccount(authorino); err != nil {
+		return err
+	} else {
```

Comment:
is the else statement needed here as if an error occur it will be returned before?

Reply:
Sorry. I don't get it. This is to return the same errors as before. It's only that the logics to get/create the service account now moved to a separate function.

Reply:
It is not a `panic`, right? So we need to propagate the error.